src/ctapipe/instrument/optics.py

```diff
@@ -263,3 +269,161 @@ def __repr__(self):
 
     def __str__(self):
         return self.name
+
+
+class PSFModel:
+    def __init__(self, **kwargs):
+        """
+        Base component to describe image distortion due to the optics of the different cameras.
+        """
+
+    @classmethod
+    def subclass_from_name(cls, name, **kwargs):
+        """
+        Obtain an instance of a subclass via its name
+
+        Parameters
+        ----------
+        name : str
+            Name of the subclass to obtain
+
+        Returns
+        -------
+        Instance
+            Instance of subclass to this class
+        """
+
+        subclasses = {base.__name__: base for base in non_abstract_children(cls)}
+
+        requested_subclass = subclasses[name]
+        return requested_subclass(**kwargs)
+
+    @abstractmethod
+    def pdf(self, *args):
+        pass
+
+    @abstractmethod
+    def update_location(self, *args):
+        pass
+
+    @abstractmethod
+    def update_model_parameters(self, *args):
+        pass
+
+
+class ComaModel(PSFModel):
+    """
+    PSF model, describing pure coma aberrations PSF effect
+    """
+
+    def __init__(
+        self,
+        asymmetry_params=[0.49244797, 9.23573115, 0.15216096],
+        radial_scale_params=[0.01409259, 0.02947208, 0.06000271, -0.02969355],
+        az_scale_params=[0.24271557, 7.5511501, 0.02037972],
+    ):
+        """
+        PSF model, describing purely the effect of coma aberration on the PSF
```

Comment:
Is there a citable reference for the math here?

Reply:
Here it is: https://www.aanda.org/articles/aa/full_html/2023/11/aa47128-23/aa47128-23.html
I will add this to the documentation

Reply:
The bib file is here:
https://github.com/cta-observatory/ctapipe/blob/main/docs/references.bib

you can then cite using:
```
:cite:p:`<bibtex key>`
```

Reply:
awesome. will do

Reply:
Which makes me realize we didn't add a reference for the hipparcos catalog

Reply:
i can add that.